ibtk/src/lagrangian/FEDataManager.cpp

```diff
@@ -826,37 +821,26 @@ FEDataManager::spread(const int f_data_idx,
     const bool use_nodal_quadrature =
         spread_spec.use_nodal_quadrature && (F_fe_type == X_fe_type && F_order == X_order);
 
-    // We only use the FECache objects if we do *not* use nodal quadrature so
-    // only perform sanity checks in that case:
-    if (!use_nodal_quadrature)
-    {
-        // This will break, at some point in the future, if we ever use
-        // nonnodal-interpolating finite elements. TODO: more finite elements
-        // will probably work.
-        std::vector<FEFamily> fe_family_whitelist{ LAGRANGE, L2_LAGRANGE };
-        TBOX_ASSERT(std::find(fe_family_whitelist.begin(), fe_family_whitelist.end(), F_fe_type.family) !=
-                    fe_family_whitelist.end());
-        TBOX_ASSERT(std::find(fe_family_whitelist.begin(), fe_family_whitelist.end(), X_fe_type.family) !=
-                    fe_family_whitelist.end());
-    }
+    // This will break, at some point in the future, if we ever use
+    // nonnodal-interpolating finite elements. TODO: more finite elements
+    // will probably work.
+    std::vector<FEFamily> fe_family_whitelist{ LAGRANGE, L2_LAGRANGE };
+    TBOX_ASSERT(std::find(fe_family_whitelist.begin(), fe_family_whitelist.end(), F_fe_type.family) !=
+                fe_family_whitelist.end());
+    TBOX_ASSERT(std::find(fe_family_whitelist.begin(), fe_family_whitelist.end(), X_fe_type.family) !=
+                fe_family_whitelist.end());
 
     if (use_nodal_quadrature)
     {
-        // Store a copy of the F vector.
-        std::unique_ptr<NumericVector<double> > F_vec_bak = F_vec.clone();
-        *F_vec_bak = F_vec;
-
         // Multiply by the nodal volume fractions (to convert densities into
         // values).
-        NumericVector<double>& F_dX_vec = F_vec;
-        const NumericVector<double>& dX_vec = *buildDiagonalL2MassMatrix(system_name);
-        F_dX_vec.pointwise_mult(F_vec, dX_vec);
-        F_dX_vec.close();
+        NumericVector<double>* dX_vec = buildGhostedDiagonalL2MassMatrix(system_name);
+        std::unique_ptr<NumericVector<double> > F_dX_vec = F_vec.clone();
+        F_dX_vec->pointwise_mult(F_vec, *dX_vec);
```

Comment:
Maybe I'm being dense but I still don't know what `F_dx` is supposed to stand for here.

Reply:
F * dX

Reply:
I will change it to `F_x_dX`.

Reply:
`F` has units of density; for nodal interaction we need to get to total amount of stuff associated with the node. `dX` is the volume associated with the node.